src/dft/backends/cufft/commit.cpp

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+* Copyright Codeplay Software Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/mkl/types.hpp"
+#include "oneapi/mkl/exceptions.hpp"
+#include "oneapi/mkl/detail/backends.hpp"
+
+#include "oneapi/mkl/dft/detail/commit_impl.hpp"
+#include "oneapi/mkl/dft/detail/types_impl.hpp"
+#include "oneapi/mkl/dft/detail/descriptor_impl.hpp"
+#include "oneapi/mkl/dft/detail/cufft/onemkl_dft_cufft.hpp"
+
+// cuFFT headers
+#include <cuda_runtime.h>
+#include <cufft.h>
+
+namespace oneapi::mkl::dft::cufft {
+namespace detail {
+
+/// Commit impl class specialization for cuFFT.
+template <dft::detail::precision prec, dft::detail::domain dom>
+class cufft_commit final : public dft::detail::commit_impl {
+private:
+    cufftHandle plan;
+
+public:
+    cufft_commit(sycl::queue& queue, const dft::detail::dft_values<prec, dom>& config_values)
+            : oneapi::mkl::dft::detail::commit_impl(queue, backend::cufft) {
+        if constexpr (prec == dft::detail::precision::DOUBLE) {
+            if (!queue.get_device().has(sycl::aspect::fp64)) {
+                throw mkl::exception("DFT", "commit", "Device does not support double precision.");
+            }
+        }
+        cufftCreate(&plan);
+
+        // The cudaStream for the plan is set a execution time so the interop handler can pick the stream.
+
+        const cufftType type = CUFFT_C2C;
+
+        auto n_copy = std::make_unique<int[]>(config_values.dimensions.size());
```

Comment:
Can we use `std::array<int, MaxSupportedDims>` since we're limited to (I think) 3D here?

Reply:
Even if we can not a std::vector here would be nicer.

Reply:
The std::array can be used so it is in https://github.com/oneapi-src/oneMKL/pull/278/commits/548a3d933e1501cbcac7135fec1beeb26724261f